internal/signalio/json.go

```diff
@@ -51,10 +50,9 @@ func (w *jsonWriter) WriteSignals(signals []signal.Set, extra ...Field) error {
 				d = make(map[string]any)
 				data[ns] = d
 			}
-			nsData, ok := d.(map[string]any)
-			if !ok {
-				return fmt.Errorf("failed to get map for namespace: %s", ns)
-			}
+			nsData := d.(map[string]any)
+			// we don't need to check for an error here (the above line) because d is always a map[string]any
```

Comment:
Might be worth extending the comment to say that "if this fails in the future the assignment to nsData below will fail"